var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -44,9 +45,16 @@ class Dyninst(Package):
     variant('stat_dysect', default=False,
             description="patch for STAT's DySectAPI")
 
+    # Dyninst depends on libelf and libdwarf prior to @9.3.0
+    # Dyninst depends on elfutils and libdwarf from @9.3.0 to but
+    # not including @develop
+    # Dyninst depends on elfutils and elfutils libdw from @develop onwards
+    # elf@0 is an abstaction for libelf
+    # elf@1 is an abstaction for elfutils
     depends_on("elf@0", type='link', when='@:9.2.99')
     depends_on("elf@1", type='link', when='@9.3.0:')
-    depends_on("libdwarf")
+    depends_on("elf@1", type='link', when='@develop')
```

Comment:
This line isn't needed as `develop` sorts above 9.3.0.

Reply:
@adamjstewart Hi Adam - I don't understand how develop sorts out above 9.3.0.  Can you explain?   Do you mean that develop is in the set that is above 9.3.0?   So I can leave off the ```
depends_on("elf@1", type='link', when='@develop')``` line?     Does the numbering scheme go: 0-9, then a-z (first characters)?   Thanks for your time!

Reply:
The sorting algorithm puts numbered releases as newer than alphabetic releases, but we have special logic in place to ensure that develop is considered newer than all other releases.

Reply:
Ok, Thanks - that makes good sense.  I will remove.